geonode/qgis_server/tests/test_views.py

```diff
@@ -133,6 +133,29 @@ def test_ogc_specific_layer(self):
         self.assertEqual(response.get('Content-Type'), 'image/tiff')
         self.assertEqual(what('', h=response.content), 'tiff')
 
+        # Layer is already on the database
+        # checking the Link
+        links = uploaded.link_set.download().filter(
+            name__in=settings.DOWNLOAD_FORMATS_RASTER)
+
+        for link in links:
+            if "download-qlr" in link.url:
+                self.assertTrue("download-qlr" in link.url)
+            if "download-qgs" in link.url:
+                self.assertTrue("download-qgs" in link.url)
```

Comment:
Just do the check like this:
```
links = uploaded.link_set.download()
qlr_link = links.get(name=link_name_for_qlr)
```

This way, we test it against the link name. If it fails, the test will fails.
Do the same for QGS